File: app/src/main/java/com/example/rwothoromo/developers/view/GithubUserProfile.java
package com.example.rwothoromo.developers.view;

import android.content.Intent;
import android.os.Build;
import android.os.Bundle;
import android.support.annotation.RequiresApi;
import android.support.v7.app.AppCompatActivity;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.example.rwothoromo.developers.R;
import com.example.rwothoromo.developers.model.GithubUser;

import java.util.Objects;

/**
 * GitHub user profile class.
 */
public class GithubUserProfile extends AppCompatActivity {

    private String githubUsername;
    private String githubUserURL;

    @RequiresApi(api = Build.VERSION_CODES.KITKAT)
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_github_user_profile);
        Objects.requireNonNull(getSupportActionBar()).setDisplayHomeAsUpEnabled(true);

        Intent intent = getIntent();
        GithubUser githubUser = intent.getParcelableExtra("githubUser");
        githubUsername = githubUser.getUsername();
        githubUserURL = githubUser.getHtmlUrl();

        ImageView profileImageView = findViewById(R.id.userProfileImageView);
        Glide.with(getApplicationContext()).load(githubUser.getAvatarUrl()).into(profileImageView);

        TextView usernameTextView = findViewById(R.id.usernameTextView);
        usernameTextView.setText(githubUsername);

        TextView profileUrlTextView = findViewById(R.id.profileUrlTextView);
        profileUrlTextView.setText(githubUserURL);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        MenuInflater inflater = getMenuInflater();
        inflater.inflate(R.menu.profile_options_menu, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {

            case android.R.id.home:
                // User chose the "Back" action, send them back to the Developer list
                finish();
                return true;

            case R.id.action_share:
                // User chose the "Share" action, launch the share intent
                Intent shareIntent = new Intent(Intent.ACTION_SEND);
                shareIntent.setType("text/plain");
                shareIntent.putExtra(Intent.EXTRA_TEXT, "Check out this awesome developer @"
                        + githubUsername + ", " + githubUserURL);
                startActivity(Intent.createChooser(shareIntent, "Share this developer"));
                return true;

            default:
                // If we got here, the user's action was not recognized.
                // Invoke the superclass to handle it.
                return super.onOptionsItemSelected(item);

        }
    }
}

File: app/src/main/java/com/rwothoromo/developers/view/GithubUserProfile.kt
package com.rwothoromo.developers.view

import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.widget.ImageView
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.bumptech.glide.Glide
import com.rwothoromo.developers.model.GithubUser
import com.rwothoromo.devfinder.R

/**
 * GitHub user profile class.
 */
class GithubUserProfile : AppCompatActivity() {

    private var githubUsername: String? = null
    private var githubUserURL: String? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_github_user_profile)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)

        val intent = intent
        val githubUser = intent.getParcelableExtra<GithubUser>("githubUser")
        githubUsername = githubUser?.username
        githubUserURL = githubUser?.htmlUrl

        val profileImageView = findViewById<ImageView>(R.id.userProfileImageView)
        Glide.with(applicationContext).load(githubUser?.avatarUrl).into(profileImageView)

        val usernameTextView = findViewById<TextView>(R.id.usernameTextView)
        usernameTextView.text = githubUsername

        val profileUrlTextView = findViewById<TextView>(R.id.profileUrlTextView)
        profileUrlTextView.text = githubUserURL
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        val inflater = menuInflater
        inflater.inflate(R.menu.profile_options_menu, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {

            android.R.id.home -> {
                // User chose the "Back" action, send them back to the Developer list
                finish()
                return true
            }

            R.id.action_share -> {
                // User chose the "Share" action, launch the share intent
                val shareIntent = Intent(Intent.ACTION_SEND)
                shareIntent.type = "text/plain"
                shareIntent.putExtra(
                    Intent.EXTRA_TEXT,
                    getString(R.string.profile_share_description, githubUsername, githubUserURL)
                )
                startActivity(
                    Intent.createChooser(
                        shareIntent,
                        getString(R.string.profile_share_title)
                    )
                )
                return true
            }

            else ->
                // If we got here, the user's action was not recognized.
                // Invoke the superclass to handle it.
                return super.onOptionsItemSelected(item)
        }
    }
}
